src/languages/cpp.js

```diff
@@ -7,6 +7,16 @@ Website: https://isocpp.org
 */
 
 function(hljs) {
+  function optional(s) {
+    return '(' + s + ')?';
```

Comment:
This function looks useful, I like the clarity of the regexps below. Don't you want to put it into the library?
But probably non-capturing group `(?:`...`)` would be better.

Reply:
I plan to add it more globally with  modules after the v10 build system hits and see how that 
looks.

> But probably non-capturing group (?:...) would be better.

Agree

Reply:
> Looks like it will interfere with #2325 though.

Hopefully git will figure it out, time will tell. :-)  I did the rename in a sep commit.